### ISSUE TITLE
snap chezmoi `2.17.2 ` doesn't start/crashes - **no** commands works

### ISSUE THREAD

**Issue by mwoz123** [twpayne/chezmoi#2165]
## Describe the bug

chezmoi doesn't work
```
marcin@dell:~$ chezmoi re-add
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /snap/chezmoi/465/chezmoi)
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /snap/chezmoi/465/chezmoi)
marcin@dell:~$ chezmoi status
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /snap/chezmoi/465/chezmoi)
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /snap/chezmoi/465/chezmoi)
marcin@dell:~$ chezmoi update
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /snap/chezmoi/465/chezmoi)
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /snap/chezmoi/465/chezmoi)
marcin@dell:~$ 
```
snap details:
```
chezmoi                  2.17.2                      465    latest/stable    twpayne          classic
````
## To reproduce

install chezmoi fron snap 
```
marcin@dell:~$ snap install chezmoi --classic
chezmoi 2.17.2 from Tom Payne (twpayne) installed
marcin@dell:~$ chezmoi update
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /snap/chezmoi/465/chezmoi)
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /snap/chezmoi/465/chezmoi)
marcin@dell:~
```

## Expected behavior

shoudn't crash. above command should work

## Output of command with the `--verbose` flag

```console
marcin@dell:~$ chezmoi status --verbose
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /snap/chezmoi/465/chezmoi)
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /snap/chezmoi/465/chezmoi)
marcin@dell:~$ ```

## Output of `chezmoi doctor`

<details>

```console
marcin@dell:~$ chezmoi doctor
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /snap/chezmoi/465/chezmoi)
/snap/chezmoi/465/chezmoi: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /snap/chezmoi/465/chezmoi)
marcin@dell:~$ 
```

</details>

## Additional context

```
marcin@dell:~$ uname -a
Linux dell 5.13.0-51-generic #58~20.04.1-Ubuntu SMP Tue Jun 14 11:29:12 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
marcin@dell:~$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04 LTS
Release:        20.04
Codename:       focal
marcin@dell:~$ 
marcin@dell:~$ ll /lib/x86_64-linux-gnu/libc.so.6
lrwxrwxrwx 1 root root 12 lut 24 20:42 /lib/x86_64-linux-gnu/libc.so.6 -> libc-2.31.so*
marcin@dell:~$ 
````

**Comment by twpayne** [comment]
Thank you for reporting this and thank you for the additional context. This is a definite bug and there are a couple of things going on:

1. chezmoi 2.17.2 go built on Ubuntu 22.04, which means that it expects a more recent version of glibc than what's in Ubuntu 20.04, i.e. binaries built on 22.04 won't run on 20.04.
2. [Snap broke authentication for snap authors](https://forum.snapcraft.io/t/snapcraft-authentication-options/30473) which means that I can't easily publish a new snap with a fix.

The immediate work-around is to stop using snaps to install chezmoi. Instead, remove the snap and use any of the other [install methods](https://www.chezmoi.io/install/). The [`chezmoi upgrade` command](https://www.chezmoi.io/reference/commands/upgrade/) should upgrade chezmoi, no matter how you installed chezmoi.

In the medium term, #2166 is an attempt to fix both the building of chezmoi on too-new Ubuntu versions and update chezmoi's build processes to the new Snap authentication workflow.

**Comment by twpayne** [comment]
Thanks for reporting this.

Fixed by #2166 and the 2.18.1 release.
